Library/Homebrew/brew.sh

```diff
@@ -744,6 +744,26 @@ fi
 source "${HOMEBREW_LIBRARY}/Homebrew/utils/analytics.sh"
 setup-analytics
 
+# Migrate from linuxbrew-core to homebrew-core
+# HOMEBREW_CORE_MERGE_MAINTAINER can be set by the maintainers
+# who merge homebrew-core into linuxbrew-core to avoid being migrated
+# We only migrate HOMEBREW_DEVELOPER users right now
+# Don't migrate on CI either
+if [[ -n "${HOMEBREW_DEVELOPER}" ]] &&
+   [[ -z "${HOMEBREW_CORE_MERGE_MAINTAINER}" ]] &&
+   [[ -z "${GITHUB_ACTIONS}" ]] &&
+   [[ "${HOMEBREW_BASH_COMMAND}" == "${HOMEBREW_LIBRARY}/Homebrew/cmd/update.sh" ]] &&
```

Comment:
Will this work only on a `brew update` or also on autoupdates?

Reply:
This should work on both. The autoupdate just calls `brew update --preinstall` so the same condition is met.

However, `update-report.rb` _won't_ be run for autoupdates which means that we'll need to figure out the best way to get the `version_scheme` updates to be shown.

Reply:
Could we run `update-report.rb` in `update.sh` when `HOMEBREW_LINUXBREW_MIGRATION` is set?

Reply:
Yes, I'm pretty sure. I've got some changes locally that I'll push in a few miuntes

Reply:
Thanks. The whole thing can be tested by setting you core remote to linuxbrew-core if needed. I can also give it a try and review any new changes :)